README.md

```diff
@@ -10,125 +10,43 @@ $ npm install p-graph
-funcs.set("putOnShoes", () => Promise.resolve("put on your shoes"));
-funcs.set("tieShoes", () => Promise.resolve("tie your shoes"));
-
-const graph = [
-  [putOnShoes, tieShoes],
-  [putOnShirt, putOnJacket],
-  [putOnShorts, putOnJacket],
-  [putOnShorts, putOnShoes],
+const nodeMap = new Map([
+  ["putOnShirt", { run:  () => Promise.resolve("put on your shirt") })],
+  ["putOnShorts", { run: () => Promise.resolve("put on your shorts")})],
+  ["putOnJacket", { run: () => Promise.resolve("put on your jacket")})],
+  ["putOnShoes", { run: () => Promise.resolve("put on your shoes")}],
+  ["tieShoes", { run: () => Promise.resolve("tie your shoes")}],
+]);
+
+const dependencies: DependencyList = [
+  // You need to put your shoes on before you tie them!
+  ["putOnShoes", "tieShoes"],
+  ["putOnShirt", "putOnJacket"],
+  ["putOnShorts", "putOnJacket"],
+  ["putOnShorts", "putOnShoes"],
 ];
 
-await pGraph(namedFunctions, graph);
+await pGraph(nodeMap, dependencies).run();
 ```
 
-3. Use a dependency map with a list of named functions
+## Concurrency Limiter
 
-```js
-const funcs = new Map();
-
-funcs.set("putOnShirt", () => Promise.resolve("put on your shirt"));
-funcs.set("putOnShorts", () => Promise.resolve("put on your shorts"));
-funcs.set("putOnJacket", () => Promise.resolve("put on your jacket"));
-funcs.set("putOnShoes", () => Promise.resolve("put on your shoes"));
-funcs.set("tieShoes", () => Promise.resolve("tie your shoes"));
-
-const depMap = new Map();
-
-depMap.set(tieShoes, new Set([putOnShoes]));
-depMap.set(putOnJacket, new Set([putOnShirt, putOnShorts]));
-depMap.set(putOnShoes, new Set([putOnShorts]));
-depMap.set(putOnShorts, new Set());
-depMap.set(putOnShirt, new Set());
-
-await pGraph(namedFunctions, graph);
-```
-
-### Using the ID as argument to the same function
-
-In many cases, the jobs that need to run are the same where the only difference is the arguments for the function. In that case, you can treat the IDs as arguments as they are passed into the Run Function.
-
-```ts
-type Id = unknown;
-type RunFunction = (id: Id) => Promise<unknown>;
-```
-
-As you can see, the ID can be anything. It will be passed as the argument for your Run Function. This is a good option if having a large number of functions inside a graph is prohibitive in memory sensitive situations.
+There are some contexts where you may want to limit the number of functions running concurrently. One example would be to prevent overloading the CPU with too many parallel tasks. The concurrency argument to `run` will limit the number of functions that start running at a given time
 
 ```js
-const funcs = new Map();
-const thatImportantTask = (id) => Promise.resolve(id);
-
-funcs.set("putOnShirt", thatImportantTask);
-funcs.set("putOnShorts", thatImportantTask);
-funcs.set("putOnJacket", thatImportantTask);
-funcs.set("putOnShoes", thatImportantTask);
-funcs.set("tieShoes", thatImportantTask);
+await pGraph(graph).run({ maxConcurrency: 3 });
```

Comment:
nit: can we use `concurrency` to be consistent with `lage`?

Reply:
Sure